src/ifcparse/IfcWrite.h

```diff
@@ -28,148 +28,149 @@
+/// for storing its value, whereas the IfcParse classes only contain
+/// lazy references to byte offsets in the IFC-SPF file.
+class IFC_PARSE_API IfcWriteArgument : public Argument {
+  public:
+    class EnumerationReference {
+      public:
+        int data;
+        const char* enumeration_value;
+        EnumerationReference(int data, const char* enumeration_value)
+            : data(data),
+              enumeration_value(enumeration_value) {}
+    };
+    class Derived {};
+    class empty_aggregate_t {};
+    class empty_aggregate_of_aggregate_t {};
+
+  private:
+    boost::variant<
+        // A null argument, it will always serialize to $
+        boost::blank,
+        // A derived argument, it will always serialize to *
+        Derived,
+        // An integer argument, e.g. 123
+
+        // SCALARS:
+        int,
+        // A boolean argument, it will serialize to either .T. or .F.
+        bool,
+        // A logical argument, it will serialize to either .T. or .F. or .U.
+        boost::logic::tribool,
+        // A floating point argument, e.g. 12.3
+        double,
+        // A character string argument, e.g. 'IfcOpenShell'
+        std::string,
+        // A binary argument, e.g. "092A" -> 100100101010
+        boost::dynamic_bitset<>,
+        // An enumeration argument, e.g. .USERDEFINED.
+        // To initialize the argument a string representation
+        // has to be explicitly passed of the enumeration value
+        // which is stored internally as an integer. The argument
+        // itself does not keep track of what schema enumeration
+        // type is represented.
+        EnumerationReference,
+        // An entity instance argument. It will either serialize to
+        // e.g. #123 or datatype identifier for simple types, e.g.
+        // IFCREAL(12.3)
+        IfcUtil::IfcBaseClass*,
+
+        // AGGREGATES:
+        empty_aggregate_t,
+        // An aggregate of integers, e.g. (1,2,3)
+        std::vector<int>,
+        // An aggregate of floats, e.g. (12.3,4.)
+        std::vector<double>,
+        // An aggregate of strings, e.g. ('Ifc','Open','Shell')
+        std::vector<std::string>,
+        // An aggregate of binaries, e.g. ("23B", "092A") -> (111011, 100100101010)
+        std::vector<boost::dynamic_bitset<>>,
+        // An aggregate of entity instances. It will either serialize to
+        // e.g. (#1,#2,#3) or datatype identifier for simple types,
+        // e.g. (IFCREAL(1.2),IFCINTEGER(3.))
+        aggregate_of_instance::ptr,
+
+        // AGGREGATES OF AGGREGATES:
+        empty_aggregate_of_aggregate_t,
+        // An aggregate of an aggregate of ints. E.g. ((1, 2), (3))
+        std::vector<std::vector<int>>,
+        // An aggregate of an aggregate of floats. E.g. ((1., 2.3), (4.))
+        std::vector<std::vector<double>>,
+        // An aggregate of an aggregate of entities. E.g. ((#1, #2), (#3))
+        aggregate_of_aggregate_of_instance::ptr>
+        container;
```

Comment:
here the closing right angle of `boost::Variant<>` should break to the next line to get a better readability. have not found the right setting yet.

Reply:
This is really an exception though. I wouldn't spend too much time fighting with this.